odoo/addons/test_read_group/tests/test_date_range.py

```diff
@@ -126,3 +126,114 @@ def test_duplicate_month(self):
         groups = self.Model.read_group(
             [('id', 'in', records.ids)], fields=['date', 'value'], groupby=['date:month', 'date:month'], lazy=False)
         self.assertEqual(groups, expected)
+
+
+class TestRelativeDateGranularity(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.Model = cls.env['test_read_group.fill_temporal']
+        cls.Model.create(({"value": "1", "date": "2021-02-09"},
+                          {"value": "20", "date": "2021-06-01"},
+                          {"value": "300", "date": "2022-02-01"}))
+
+    def test_full_usecase_month(self):
+        result = self.Model.read_group([],
+                                       fields=['date', 'value'],
+                                       groupby=['date:month_number'])
+        self.assertEqual(result, [
+            {
+                'date:month_number': 2.0,
+                'date_count': 2,
+                'value': 301,
+                '__domain': [('date.month_number', '=', 2.0)]
+            }, {
+                'date:month_number': 6.0,
+                'date_count': 1,
+                'value': 20,
+                '__domain': [('date.month_number', '=', 6.0)]
+            }
+        ])
+        res = self.Model.search(result[0]['__domain'])
+        self.assertEqual(len(res), 2)
+        self.assertEqual(res.mapped('value'), [1, 300])
+
+    def test_full_usecase_week(self):
+        result = self.Model.read_group([],
+                                       fields=['date', 'value'],
+                                       groupby=['date:week_number'])
+        self.assertEqual(result, [
+            {
+                'date:week_number': 5.0,
+                'date_count': 1,
+                'value': 300,
+                '__domain': [('date.week_number', '=', 5.0)]
+            }, {
+                'date:week_number': 6.0,
+                'date_count': 1,
+                'value': 1,
+                '__domain': [('date.week_number', '=', 6.0)]
+            }, {
+                'date:week_number': 22.0,
+                'date_count': 1,
+                'value': 20,
+                '__domain': [('date.week_number', '=', 22.0)]
+            }])
```

Comment:
One of the recommended ways to indent Python code is
```py
some_call([
    ...  # indent 4 spaces w.r.t. the line that introduces the opening bracket
])  # the closing bracket is indented like the line that introduces the opening bracket
```
so I suggest
```suggestion
            },
        ])
```